docs/guides/building/arm-cross-compile.md

```diff
@@ -0,0 +1,162 @@
+# Cross-Compiling Tauri Applications for ARM-based Devices
+
+This guide explains how to cross-compile your Tauri application for ARM-based devices, such as the Raspberry Pi. Compiling directly on the device may not be feasible due to limited RAM, so we'll explore two methods: manual compilation using Linux or WSL on Windows and automatic cross-compilation using a GitHub Action.
+
+## Manual Compilation
+
+Manual compilation is suitable when you don't need to compile your application frequently and prefer a one-time setup. Follow the steps below:
+
+### Prerequisites
+
+:::warning Important Note
+
+Ubuntu 20.04 WSL has issues with webkit dependencies. It's recommended to use Debian instead.
+
+:::
+
+- For Windows, install Debian on WSL using the [wsl setup guide].
+- On Linux, ensure your machine has the same GLIBC version as the target device. Check the version using: `ldd --version`.
+- Use a Linux distribution based on Debian/Ubuntu for this guide, as the commands shown use the `apt` package manager and have been tested on Debian 11.
+
+### Initial Setup
+
+Before cross-compiling your app, ensure all required tools for building your application are installed:
+
+1. Update the distro: `sudo apt-get update && sudo apt-get upgrade -y`
+2. Add Tauri [system requirements].
+3. Install Rust and ensure it's up-to-date: `rustup update`.
+4. Install [Tauri using Cargo]: `cargo install tauri-cli`.
+5. Install necessary tools for your frontend framework, like npm and Next.js.
+
+:::info OpenSSL Workaround for WSL on Windows
+
+If using WSL on Windows and encountering OpenSSL-related errors, add the following line to `Cargo.toml` in `<project-root>/src-tauri/Cargo.toml` within the `[dependencies]` section:
+```toml
+openssl-sys = {version = "0.9.66", features = ["vendored"]}
+```
+:::
+
+### Cross-Compiling
```

Comment:
We need to explain arm 7 vs 8/64 in this section imo (and add instructions for the latter). Both are equivally often requested, arm 7 for (older) Pis and arm64 for more desktop-like targets.

Reply:
Should I add a second command showing the difference to each step, or a separate step-by-step guide entirely? Should I also discuss the background to both architectures and what is compatible with which?
I also don't think I can properly test it, as I only have Raspberry Pi 3 and 4.

Reply:
> Should I add a second command showing the difference to each step, or a separate step-by-step guide entirely?

a seperate guide sounds too much, i'd either just mention the differences, or if that's not clear enough in your opinion, make use of the tabs component.

> I also discuss the background to both architectures and what is compatible with which?

You _can_ but it's not a requirement in my eyes. (keep it short though if possible :D )

> I also don't think I can properly test it, as I only have Raspberry Pi 3 and 4.

no worries, i didn't hear any complains about my arm64 instructions yet, and either way we can just fix the docs later if it ends up being wrong in some way.

Reply:
I added the separate commands for armv8, let me know if it's ok

Reply:
cool, looks good!